[Reviewer: Andy] add capability to restrict which processes receive ato mapping

johadalin opened the pull request:
Adding in function to find the process id, then get its name. This can then be compared against the list provided at the top of the file, and only return the user mapping if it is a whitelisted process. otherwise it  will return that the user was not found. 

This should avoid the problem of all possible users existing when trying to add/modify a user etc.

bossmc commented on libnss_ato.c:
```diff
@@ -27,6 +27,26 @@
+ */
+const char *VALID_PROC_NAMES[] = {"sshd",
+                                  "login",
+                                  "w"};
```
How was this short-list determined?  Are we confident it's correct?  How can we check?

Would it make sense to log out the binary (and whether we're being used/skipped) every time to make this easier to audit?

bossmc replied:
Also would it be better if this list were loaded from a config file rather than hard-coded?

johadalin replied:
The short list was taken direct from the ABU version, which has been around for a couple of years, so obvious bugs should have been found by now.
It could be better in terms of flexibility to pull it from config files, but i'd say that is something for future enhancement. given that this is just for authentication via radius, we dont really want other processes taking user mappings, i think.

bossmc commented on libnss_ato.c:
```diff
@@ -89,6 +109,57 @@ get_static(char **buffer, size_t *buflen, int len)
+  int i;
+  for (i = 0; i < arraylen(VALID_PROC_NAMES); i++) {
+    if (strcmp(name, VALID_PROC_NAMES[i]) == 0) {
+      // DEBUG_LOG("libnss_ato: Process name matched '%s'", VALID_PROC_NAMES[i]);
```
Don't leave commented-out lines lying around, uncomment or delete.

bossmc commented:
Reviewed.

johadalin commented:
removed 'who' from the whitelist. 
Added a changelog entry. should certainly do for the first version. 
I've left the non-working syslog attempts in there for now, and am going to raise an issue against them, and against the hard-coded whitelist.

Are you happy with the changelog format?

bossmc commented:
Cool
